### DISCUSSION TITLE
Casting with EnableNestedPaths

### DISCUSSION THREAD

**Discussion by ghost** [OData/AspNetCoreOData#466] [Q&A]
**I have an entity :** 

```
public class DataSource
    {
        [Key]
        public string Name { get; set; }
        public ProductType Type { get; set; }     
    }
```

**And a derived Class** 

```
 public class EnggDataSource : DataSource
    {
        public List<Simulation> Simulations { get; set; }
        public EnggDataSource()
        {
           Simulations = new List<Simulation> { new Simulation { Name = "sim 1" }, new Simulation { Name = "sim 2" } };
        }
    }

public enum ProductType
    {
        Engineering,     
        Excel
    }

    public class Simulation
    {
        [Key]
        public string Name { get; set; }
    }
```

**And a controller** 

```
public class DatasourcesController : ODataController
    {
        private List<DataSource> _dataSources = new List<DataSource>()
        {
            new EnggDataSource()
            {
                Name = "E3d1_ds",
                Type = ProductType.Engineering,
                 Simulations = new List<Simulation> { new Simulation { Name = "sim 1" }, new Simulation { Name = "sim 2" } }
            },
        };

        [EnableQuery]
        [EnableNestedPaths]
        [HttpGet("Datasources")]
        public ActionResult Get()
        {
           return  Ok(_dataSources.AsQueryable());
        }
}
```

**And EDMModel**

```
private static IEdmModel GetEdmModel()
        {
            var builder = new ODataConventionModelBuilder();
            builder.EntitySet<DataSource>("Datasources");
            return builder.GetEdmModel();
         }
```

**Question**
If I want to get the Simulations property under the EnggDataSource entity with Enable nested paths 
I am trying 
http://localhost:60339/odata/Datasources('E3d1_ds')/ODataService.Models.EnggDataSource/Simulations

Which does not work

and neither does this 
http://localhost:60339/odata/Datasources('E3d1_ds')/Simulations

But if there were no derived entity and Simulations was part of Datasources directly I have no issues querying simulations ...

Is this a flaw or is it by design ?

**Comment by xuzhg** [comment]
Are you using the 'EnableNestedPaths` and using 7.x version?  

EnableNestedPaths is not supported in 8.x because it's totally different routing mechanism. 

I'd like to invite @habbes to take a look for it. In my opinion, it seems a feature gap.

**Reply by ghost** [reply]
Yes @xuzhg  I am using the latest 7.x release

Entity framework supports [inheritance](https://docs.microsoft.com/en-us/ef/core/modeling/inheritance), so I assumed nestedpaths should too. Thanks for the reply... atleast I know my issue has been heard :)

**Reply by ghost** [reply]
@xuzhg  @habbes  Can you please let me know if there is any update on this ? Thanks for your time.
